packages/alfa-parser/src/parser.ts

```diff
@@ -343,6 +344,26 @@ export namespace Parser {
     );
   }
 
+  /**
+   * Parse the first element in the input if it satisfies the given refinement.
+   *
+   * @remarks
+   * The function only works for scenarios where the input is of type `Slice`.
+   */
+  export function parseIf<I, T extends I>(
+    refinement: Refinement<I, T>
+  ): Parser<Slice<I>, T, string> {
+    return (input) =>
+      input
+        .first()
+        .map((token) =>
+          refinement(token)
+            ? Result.of<[Slice<I>, T], string>([input.rest(), token])
+            : Err.of("Mismatching token")
+        )
+        .getOr(Err.of("No token left"));
+  }
+
```

Comment:
A `Predicate<T>` is just a `Refinement<T, T>`, so that should work out of the box.

Reply:
True.
This is however fairly easy, since it is only `input => input.first().map(first => Ok.of([input.rest(), first])).getOr(Err.of("Empty input"))`.
If needed, it might be better to add a `Slice#split` helper that does the `input.first().map(first => [input.rest(), first])` bit, and then just call `input.split().map(Ok.of).getOr(Err.of("Empty input"))`. But that still feels like putting a bit too more dependency inside `Slice` (it only cares about `split` because of our usage of `Slice` is in `Parser` that needs a `[I, T]` output).

Reply:
Yeah, and it was only a few places after all, so I have just duplicated it for now, then we can revisit later.